src/trace/listener.ts

```diff
@@ -131,8 +139,14 @@ export class TraceListener {
 
     this.lambdaSpanParentContext = this.inferredSpan?.span || parentSpanContext;
     this.context = context;
-    this.triggerTags = extractTriggerTags(event, context);
+    const eventSource = parseEventSource(event);
+    this.triggerTags = extractTriggerTags(event, context, eventSource);
     this.stepFunctionContext = StepFunctionContextService.instance().context;
+
+    const result = getSpanPointerAttributes(eventSource, event);
+    if (result) {
+      this.spanPointerAttributesList.push(...result);
```

Comment:
I'm not completely sure how our wrapping works, but I just want to confirm. It looks to me like `this.spanPointerAttributesList` is an instance attribute. Yet because we only create a single trace listener, then this list is never reset, and will just grow in size.

I'm also wondering, separately, if instead of calling `.push` on the list if we can just straight up set `this.spanPointerAttributesList = getSpanPointerAttributes(eventSource, event);`?

Reply:
You're right, thank you for catching this. Yes, we should be able to just set `this.spanPointerAttributesList` to the result.